dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentTypeLoader.java

```diff
@@ -0,0 +1,303 @@
+import com.dotcms.contenttype.model.field.TextAreaField;
+import com.dotcms.contenttype.model.field.TextField;
+import com.dotcms.contenttype.model.field.TimeField;
+import com.dotcms.contenttype.model.field.WysiwygField;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.List;
+
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+/**
+ * @author will
+ */
+public class ContentTypeLoader implements DotLoader {
+
+    private long fakeIdentifier = Long.MAX_VALUE;
+    private long fakeInode = Long.MAX_VALUE;
+    private String fakeTitle = "Content Title";
+
+
+
+    public void invalidate(ContentType contentType, PageMode mode) {
+        removeContentTypeFile(contentType);
+    }
+
+
+
+    public InputStream buildVelocity(ContentType type, String filePath, PageMode mode ) {
+
+        // let's write this puppy out to our file
+        StringBuilder sb = new StringBuilder();
+
+        // CONTENTLET CONTROLS BEGIN
+        sb.append("#set( $EDIT_CONTENT_PERMISSION =$EDIT_CONTENT_PERMISSION")
+            .append(fakeInode)
+            .append(" )");
+        sb.append("#set( $CONTENT_INODE ='")
+            .append(fakeInode)
+            .append("' )");
+        sb.append("#set( $IDENTIFIER_INODE ='")
+            .append(fakeIdentifier)
+            .append("' )");
+
+        // set all properties from the contentlet
+        sb.append("#set( $ContentInode ='")
+            .append(fakeInode)
+            .append("' )");
+        sb.append("#set( $ContentIdentifier ='")
+            .append(fakeIdentifier)
+            .append("' )");
+        sb.append("#set( $ContentletTitle =\"")
+            .append(UtilMethods.espaceForVelocity(fakeTitle))
+            .append("\" )");
+
+        // Structure fields
+        List<Field> fields = type.fields();
+
+
+        for (Field field : fields) {
+
+
+            String contField = field.dbColumn();
+            String contFieldValue = null;
+            Object contFieldValueObject = null;
+            if (UtilMethods.isSet(contField)) {
+                try {
+                    contFieldValueObject = field.name();
+                    contFieldValue = contFieldValueObject == null ? "" : contFieldValueObject.toString();
+                } catch (Exception e) {
+                    Logger.error(this.getClass(), "writeContentletToFile: " + e.getMessage());
+                }
+                if (!(field instanceof DateTimeField || field instanceof DateField || field instanceof TimeField)) {
+                    sb.append("#set( $")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "#set( $" 24 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)